Sources/BSON/ExtendedJSONDecoder.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+/// Decoder for ExtendedJSON
+public class ExtendedJSONDecoder {
+    /// Initialize a Decoder
+    public init() {
+        fatalError("unimplemented")
+    }
+
+    /// Takes in JSON as `Data` encoded with `.utf8` and runs it through a `JSONDecoder` to get an
+    /// instance of the `JSON` enum. Then a `BSON` enum instance is created via the `JSON`.
+    /// The `BSON is then passed through a `BSONDecoder` where it is outputted as a `T`
+    ///
+    /// Data --> JSON --> BSON --> T
+    ///
+    /// - Parameters:
+    ///   - type: Codable type to decode the input into.
+    ///   - data: `Data` which represents the `JSON` that will be decoded.
```

Comment:
while `JSON` is actually a type, it's not one that the user knows about, so I think we can omit the backticks here and the next line. otherwise they might think there is some type they are supposed to be passing in here rather than a `Data`.

Reply:
done